docs/chapters/03_booster-architecture.md

```diff
@@ -795,3 +795,132 @@ Booster GraphQL API also provides support for real-time updates using subscripti
 
 The Booster GraphQL API provides support for filtering Read Models on `queries` and `subscriptions`. To get more information about it go to the [GraphQL API](chapters/04_features#filtering-a-read-model) section.
+
+#  5. Queries
+
+ReadModels offer read operations over reduced events. In the other hand, Queries provide an interface for custom read operations. They are included in the _Read API_.
+
+Queries are classes decorated with the `@Query` decorator that have a `handle` method.
+
+```typescript
+import { Booster, Query } from '@boostercloud/framework-core'
+import { CartReadModel } from '../read-models/cart-read-model'
+
+@Query({
+  authorize: 'all',
+})
+export class CartsByCountry {
+  public constructor() {}
+
+  public static async handle(query: CartsByCountry): Promise<Record<string, Array<CartReadModel>>> {
+    const carts: Array<CartReadModel> = (await Booster.readModel(CartReadModel)
+      .filter({
+        shippingAddress: {
+          country: {
+            isDefined: true,
+          },
+        },
+      })
+      .paginatedVersion(false)
+      .search()) as Array<CartReadModel>
+
+    return carts.reduce((group: Record<string, Array<CartReadModel>>, cartReadModel: CartReadModel) => {
+      const country = cartReadModel.shippingAddress?.country || ''
+      group[country] = group[country] ?? []
+      group[country].push(cartReadModel)
+      return group
+    }, {})
+  }
+}
+```
+
+##  Queries naming convention
+
+We recommend use the `Query` suffix in your queries name.
+
+Despite you can place your queries in any directory, we strongly recommend you to put them in `<project-root>/src/queries`.
+
+```text
+<project-root>
+├── src
+│   ├── commands
+│   ├── common
+│   ├── config
+│   ├── entities
+│   ├── read-models
+│   ├── events
+│   ├── queries      <------ put them here
+│   └── index.ts
+```
+
+##  Creating a query
+
+The preferred way to create a query is by using the generator, e.g.
+
+```shell
+boost new:query ItemsInCountry --fields country:string
+```
+
+The generator will create a Typescript class under the queries directory `<project-root>/src/queries/items-in-country.ts`.
+
+Queries classes can also be created by hand and there are no restrictions. The structure of the data is totally open and can be as complex as you can manage in your projection functions.
+
+## The query handler function
+
+Each query class must have a method called `handle`. This function is the command handler, and it will be called by the framework every time one instance of this query is submitted. Inside the handler you can run validations, return errors and query entities to make decisions.
+
+### Returning a value
+
+By default, the command handler function is generated with `Promise<void>` as a return type,
+and in consequence, it will return `true` when called through the GraphQL.
+
+> [!NOTE] `true` is returned because GraphQL lacks of a `void` type, and it represents that the execution was successful.
```

Comment:
Copy paste issue. I'm not sure if this section is needed because a query without a return value doesn't make sense. Maybe saying something along those lines would be ok